indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceIngestionSpec.java

```diff
@@ -47,7 +48,8 @@ public DatasourceIngestionSpec(
       @JsonProperty("filter") DimFilter filter,
       @JsonProperty("granularity") QueryGranularity granularity,
       @JsonProperty("dimensions") List<String> dimensions,
-      @JsonProperty("metrics") List<String> metrics
+      @JsonProperty("metrics") List<String> metrics,
+      @JsonProperty("ignoreWhenNoSegments") Boolean ignoreWhenNoSegments
```

Comment:
`boolean` will default to false if not present, so we don't need to use `Boolean`